manifests/plugin/remote_execution/ssh.pp

```diff
@@ -54,40 +54,44 @@
     listen_on => $listen_on,
   }
 
-  if $ssh_kerberos_auth {
-    $kerberos_pkg = "${foreman_proxy::params::ruby_package_prefix}net-ssh-krb"
-    package { $kerberos_pkg:
-      ensure => present,
+  if $mode =~ /ssh/ {
+    if $ssh_kerberos_auth {
+      $kerberos_pkg = "${foreman_proxy::params::ruby_package_prefix}net-ssh-krb"
+      package { $kerberos_pkg:
+        ensure => present,
+      }
     }
-  }
 
-  if $generate_keys {
-    file { $ssh_identity_dir:
-      ensure => directory,
-      owner  => $foreman_proxy::user,
-      group  => $foreman_proxy::user,
-      mode   => '0700',
-    }
-    -> exec { 'generate_ssh_key':
-      command => "${ssh_keygen} -f ${ssh_identity_path} -N '' -m pem",
-      user    => $foreman_proxy::user,
-      cwd     => $ssh_identity_dir,
-      creates => $ssh_identity_path,
-    }
-    if $install_key {
-      # Ensure the .ssh directory exists with the right permissions
-      file { '/root/.ssh':
+    if $generate_keys {
+      file { $ssh_identity_dir:
         ensure => directory,
-        owner  => 'root',
-        group  => 'root',
+        owner  => $foreman_proxy::user,
+        group  => $foreman_proxy::user,
         mode   => '0700',
       }
-      -> exec { 'install_ssh_key':
-        path    => '/usr/bin:/usr/sbin:/bin',
-        command => "cat ${ssh_identity_path}.pub >> /root/.ssh/authorized_keys",
-        unless  => "grep -f ${ssh_identity_path}.pub /root/.ssh/authorized_keys",
-        require => Exec['generate_ssh_key'],
+      -> exec { 'generate_ssh_key':
+        command => "${ssh_keygen} -f ${ssh_identity_path} -N '' -m pem",
+        user    => $foreman_proxy::user,
+        cwd     => $ssh_identity_dir,
+        creates => $ssh_identity_path,
+      }
+      if $install_key {
+        # Ensure the .ssh directory exists with the right permissions
+        file { '/root/.ssh':
+          ensure => directory,
+          owner  => 'root',
+          group  => 'root',
+          mode   => '0700',
+        }
+        -> exec { 'install_ssh_key':
+          path    => '/usr/bin:/usr/sbin:/bin',
+          command => "cat ${ssh_identity_path}.pub >> /root/.ssh/authorized_keys",
+          unless  => "grep -f ${ssh_identity_path}.pub /root/.ssh/authorized_keys",
+          require => Exec['generate_ssh_key'],
+        }
       }
     }
+  } elsif $mode == 'pull-mqtt' {
```

Comment:
Line 43 in this class will need to be updated to include this new value. @adamruzicka is this the nomenclature you'd prefer to have? or just pull? or just mqtt?

Reply:
`pull-mqtt` is correct. `pull` is also an option, but then it wouldn't use mqtt. All the available modes are listed here https://github.com/theforeman/smart_proxy_remote_execution_ssh/blob/master/settings.d/remote_execution_ssh.yml.example#L8